### PR TITLE
feat(api): use tv-pairing-code service

### PR DESCRIPTION
**Pull request by stdevi**
## Description of the change

This PR changes api so that `tv/code` and `tv/code/{code}` requests are sent to the new tv pairing code service.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (Changed implementation, but existing functionality and APIs are not changed)

### Code review 

- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: Sources/QminderAPI.swift [hunk 1]
```diff
@@ -10,268 +10,324 @@ import Foundation
...
-      completion(result)
+    
+    /// Qminder API key
+    private var apiKey: String?
+    
+    /// Qminder API address
+    private var serverAddress: String
+    
+    /// Qminder TV Pairig Code address
+    private var tvPairingServerAddress: String
```
Comment (stdevi) [inline Sources/QminderAPI.swift:21]: Added tvPairingServerAddress

File: Sources/QminderAPI.swift [hunk 2]
```diff
@@ -10,268 +10,324 @@ import Foundation
...
+    
+    /// Queue to return result in
+    private var queue = DispatchQueue.main
+    
+    public init(apiKey: String? = nil,
+                serverAddress: String = "https://api.qminder.com/v1",
+                tvPairingServerAddress: String = "https://tv.qminder.com") {
```
Comment (stdevi) [inline Sources/QminderAPI.swift:28]: Set the value of `tvPairingServerAddress`

File: Sources/QminderAPI.swift [hunk 3]
```diff
@@ -10,268 +10,324 @@ import Foundation
...
-  
-  public func pairTV(code: String,
-                     secret: String,
-                     completion: @escaping (Result<TVAPIData, QminderError>) -> Void) {
-    pairTV(code: code, secret: secret) { result, _ in
-      completion(result)
+    
+    public func getPairingCodeAndSecret(
+        completion: @escaping (Result<TVPairingCode, QminderError>, HTTPURLResponse?) -> Void) {
+            fetchTvPairing(.tvCode, decodingType: TVPairingCode.self) { result, response in
```
Comment (stdevi) [inline Sources/QminderAPI.swift:106]: It calls `fetchTvPairing` now

File: Sources/QminderAPI.swift [hunk 4]
```diff
@@ -10,268 +10,324 @@ import Foundation
...
+                                        _ completion: @escaping (Result<T, QminderError>, HTTPURLResponse?) -> Void) {
+        performTvPairingRequestWith(endPoint) { result, response in
+            switch result {
+            case let .success(data):
+                completion(data.decode(decodingType), response)
+            case let .failure(error):
+                completion(Result(error), response)
+            }
+        }
     }
```
Comment (stdevi) [inline Sources/QminderAPI.swift:228]: Added this function

File: Sources/QminderAPI.swift [hunk 5]
```diff
@@ -10,268 +10,324 @@ import Foundation
...
+                               httpURLResponse)
+                }
+            }.resume()
+        } catch {
+            completion(Result(error.qminderError), nil)
         }
-      }.resume()
-    } catch {
-      completion(Result(error.qminderError), nil)
     }
```
Comment (stdevi) [inline Sources/QminderAPI.swift:304]: Added this function

File: Sources/QminderAPI.swift [hunk 6]
```diff
@@ -16,13 +16,19 @@ public struct QminderAPI: QminderAPIProtocolResponse {
   
   /// Qminder API address
   private var serverAddress: String
+
+  /// Qminder TV Pairig Code address
```
Comment (2ndalpha) [inline Sources/QminderAPI.swift:20]: Typo: Pairig

**Review by 2ndalpha — CHANGES_REQUESTED** [review]

File: Sources/QminderAPI.swift (hunk 6 above)
Comment (stdevi) [reply Sources/QminderAPI.swift]: Fixed

**Review by 2ndalpha — APPROVED** [review]
🚀

**Pull request merged**